client/rest/handlers.go

```diff
@@ -145,101 +142,46 @@ type Context struct {
 
 func (ctx *Context) RegisterHandlers(r *mux.Router) error {
 	ctx.Keys.Register(r)
-	r.HandleFunc("/build/send", doSend).Methods("POST")
 	r.HandleFunc("/sign", doSign).Methods("POST")
-	r.HandleFunc("/tx", doPostTx).Methods("POST")
-	r.HandleFunc("/query/account/{signature}", doAccountQuery).Methods("GET")
 
 	return nil
 }
 
-func extractAddress(signature string) (address string, err *ErrorResponse) {
-	// Expecting the signature of the form:
-	//  sig:<ADDRESS>
-	splits := strings.Split(signature, ":")
-	if len(splits) < 2 {
-		return "", &ErrorResponse{
-			Error: `expecting the signature of the form "sig:<ADDRESS>"`,
-			Code:  406,
-		}
-	}
-	if splits[0] != "sigs" {
-		return "", &ErrorResponse{
-			Error: `expecting the signature of the form "sig:<ADDRESS>"`,
-			Code:  406,
-		}
-	}
-	return splits[1], nil
-}
-
-func doAccountQuery(w http.ResponseWriter, r *http.Request) {
-	query := mux.Vars(r)
-	signature := query["signature"]
-	address, errResp := extractAddress(signature)
-	if errResp != nil {
-		writeCode(w, errResp, errResp.Code)
-		return
-	}
-	actor, err := commands.ParseActor(address)
-	if err != nil {
-		writeError(w, err)
-		return
-	}
-	actor = coin.ChainAddr(actor)
-	key := stack.PrefixedKey(coin.NameCoin, actor.Bytes())
-	account := new(coin.Account)
-	proof, err := proofs.GetAndParseAppProof(key, account)
-	if lightclient.IsNoDataErr(err) {
-		err := fmt.Errorf("account bytes are empty for address: %q", address)
-		writeError(w, err)
-		return
-	} else if err != nil {
-		writeError(w, err)
-		return
-	}
-
-	if err := proofs.OutputProof(account, proof.BlockHeight()); err != nil {
-		writeError(w, err)
-		return
-	}
-	writeSuccess(w, account)
-}
-
 func doPostTx(w http.ResponseWriter, r *http.Request) {
 	tx := new(basecoin.Tx)
-	if err := parseRequestJSON(r, tx); err != nil {
-		writeError(w, err)
+	if err := common.ParseAndValidateRequestJSON(r, tx); err != nil {
+		common.WriteError(w, err)
 		return
 	}
 	commit, err := PostTx(*tx)
 	if err != nil {
-		writeError(w, err)
+		common.WriteError(w, err)
 		return
 	}
 
-	writeSuccess(w, commit)
+	common.WriteSuccess(w, commit)
 }
 
 func doSign(w http.ResponseWriter, r *http.Request) {
```

Comment:
Sorry, I think I tagged the wrong function... scroll down to `doSend`.  There are no references to it in the `client/rest` package.